signup/api/src/java/org/sakaiproject/signup/logic/SignupUser.java

```diff
@@ -122,36 +123,27 @@ public int hashCode() {
 		return eid.hashCode();
 	}
 
+	private String getSortName() {
+		String sortName = StringUtils.isNotBlank(lastName) ? lastName : "";

```

Comment:
why can't we use UserDirectoryService.getSortName instead of re-implementing it here?

Reply:
It's not part of the api signature. It's part of BaseUserEdit and SignupUser doesn't extend that, or compose it.

Reply:
I see the two points but I just pushed changes implemeting the @ottenhoff approach. I understand that if in the future the user's sorting behaviour changes, this is going to be another change point so that way we get rid of stay changing in this point. Anyway, if reimplementing the compareTo in SignupUser in the way that BaseUserDirectoryService is the better option, I have no problem and it is a fast change.

Reply:
In the last commit I removed the compareTo and I used the sorting lists returned or sorted by the UserDirectoryService, so if we create a new list using the first, the second list is going to be sorted.

Reply:
Fine by me. It's a shame SignupUser doesn't extend or compose User in some way so we could sort SignupUser anywhere, not by having to sort the User objects first, but that would be a big change, a really big change I'd imagine.